### ISSUE TITLE
Leverage `finalizer` for `Package` deletions

### ISSUE THREAD

**Issue by mjnagel** [defenseunicorns/uds-core#523]
### Describe what should be investigated or refactored

In the current state of things deletions happen in a watch in our pepr code. As we manage additional "external resources" (such as keycloak clients), blocking on that deletion is critical to prevent errors when a rapid delete + create process happens. Kubernetes finalizers feel like the right solution to ensure that someone cannot reuse a `Package` name before the previous one is fully cleaned up.

### Links to any relevant code

This block of code is the most relevant - note that these are two separate flows and there is currently no "block on delete" in here:

https://github.com/defenseunicorns/uds-core/blob/d71d83ed4d6e6a35724e70fc5a27cb7ff6e1adaa/src/pepr/operator/index.ts#L43-L60

### Additional context

https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/

**Comment by mjnagel** [comment]
Linking Pepr issue/feature suggestion for native finalizer support - https://github.com/defenseunicorns/pepr/issues/1022

**Comment by mjnagel** [comment]
May put this on hold to see what shakes out from the Pepr native support feature request.

**Comment by mjnagel** [comment]
Still pending upstream work on Pepr to natively support this - closed the WIP PR for now as the upstream implementation may change how that looks.

**Comment by mjnagel** [comment]
Unblocked as pepr now [natively supports this](https://docs.pepr.dev/main/user-guide/actions/finalize/) - reference the closed PR for a draft implementation: https://github.com/defenseunicorns/uds-core/pull/634
